lib/Sema/CSRanking.cpp

```diff
@@ -446,6 +451,32 @@ static bool isDeclAsSpecializedAs(TypeChecker &tc, DeclContext *dc,
         return inProtocolExtension2;
       }
 
+      // A concrete type member is always more specialised than a protocol
+      // member (bearing in mind that we have already handled the case where
+      // exactly one member is in a protocol extension). Only apply this rule in
+      // Swift 5 mode to better maintain source compatibility under Swift 4
+      // mode.
+      //
+      // Don't apply this rule when comparing two overloads found through
+      // dynamic lookup to ensure we keep cases like this ambiguous:
+      //
+      //    @objc protocol P {
+      //      var i: String { get }
+      //    }
+      //    class C {
+      //      @objc var i: Int { return 0 }
+      //    }
+      //    func foo(_ x: AnyObject) {
+      //      x.i // ensure ambiguous.
+      //    }
+      //
+      if (true && !isDynamicOverloadComparison) {
```

Comment:
It should probably be simplified to `if (!isDynamicOverloadComparison)`

Reply:
@xedin I removed the version checks temporarily to allow the compat suite to exercise the Swift 5 path in 685ec32919d6c2efa4cbe9219d9c50857bd61482. `true` should be `tc.Context.isSwiftVersionAtLeast(5)`, and below, `false` should be `!tc.Context.isSwiftVersionAtLeast(5)`.

Reply:
Ah, gotcha!